custom_components/samsungtv_tizen/logo.py

```diff
@@ -0,0 +1,209 @@
+
+        self._last_check = None
+
+        self._logo_file_path = (
+            os.path.dirname(os.path.realpath(__file__)) + LOGO_FILE_PATH
+        )
+        self._logo_file_download_path = (
+            os.path.dirname(os.path.realpath(__file__)) + LOGO_FILE_DOWNLOAD_PATH
+        )
+
+    def set_logo_color(self, logo_option):
+        """ Sets the logo color option and image base url if not already set to this option """
+        logo_option = (
+            LOGO_OPTIONS_MAPPING[LogoOption(logo_option)]
+            if logo_option
+            else LOGO_OPTION_DEFAULT[1]
+        )
+        if self._logo_option != logo_option:
+            _LOGGER.debug("Setting logo option to %s", logo_option)
+            if logo_option in LOGO_OPTIONS_MAPPING.values():
+                if logo_option == "none":
+                    self._media_image_base_url = None
+                else:
+                    self._media_image_base_url = "{}export/{}".format(
+                        LOGO_BASE_URL, logo_option
+                    )
+                self._logo_option = logo_option
+            else:
+                _LOGGER.warning(
+                    "Unrecognized value '%s' for 'Display logos' option. Using default value.",
+                    logo_option,
+                )
+                self._logo_option = LOGO_OPTION_DEFAULT[1]
+                self._media_image_base_url = "{}export/{}".format(
+                    LOGO_BASE_URL, LOGO_OPTION_DEFAULT[1]
+                )
+
+    async def _async_ensure_latest_path_file(self):
+        """ Does check if logo paths file exists and if it does - is it out of date or not. """
+        if self._media_image_base_url is not None and (
+            self._last_check is None
+            or self._last_check
+            < (
+                datetime.now().astimezone()
+                - timedelta(days=LOGO_FILE_DAYS_BEFORE_UPDATE)
+            )
+        ):
+            if os.path.isfile(self._logo_file_download_path):
+                file_date = datetime.utcfromtimestamp(
+                    os.path.getmtime(self._logo_file_download_path)
+                ).astimezone()
+                if file_date < (
+                    datetime.now().astimezone()
+                    - timedelta(days=LOGO_FILE_DAYS_BEFORE_UPDATE)
+                ):
+                    try:
+                        async with self._session.head(
+                            LOGO_BASE_URL + "logo_paths.json"
+                        ) as response:
+                            url_date = datetime.strptime(
+                                response.headers.get("Last-Modified"),
+                                "%a, %d %b %Y %X %Z",
+                            ).astimezone()
+                            if url_date > file_date:
+                                self._last_check = datetime.now().astimezone()
+                                await self._download_latest_path_file()
+                    except:
```

Comment:
It is possible to use a more specific exception?

Reply:
Done!